packages/doc/src/components/CodeBlock/PrismHighlight/PrismHighlight.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useContext } from 'react';
+import { string, bool } from 'prop-types';
+import Highlight, { defaultProps } from 'prism-react-renderer';
+import githubTheme from 'prism-react-renderer/themes/github';
+import styled from 'styled-components';
+import { LiveEditor } from 'react-live';
+
+import CodeBlockContext from '../CodeBlockContext';
+
+const defaultPropsWithTheme = {
+  ...defaultProps,
+  theme: githubTheme,
+};
+
+const Pre = styled.pre`
+  ${({
+    bordered,
+    theme: {
+      yoga: {
+        colors: { primary: primaryPallete, gray: grayPallete },
+      },
+    },
+  }) => `
+    border-radius: 5px;
+    font-size: 15px;
+    margin: 0;
+    padding: 18px;
+
+    .token.string {
+      color: ${primaryPallete[3]} !important;
+    }
+
+    overflow: auto;
+    ${bordered ? `border: 1px solid ${grayPallete[3]};` : ''}
+
+    @media (max-width: 900px) {
+      font-size: 12px;
+    }
+
+    div pre {
+        padding: 0 !important;
+    }
+  `}
+`;
+
+const PrismHighlight = ({ code, liveEditor }) => {
+  const highlightCode = !code ? useContext(CodeBlockContext).code : code;
```

Comment:
```suggestion
  const highlightCode = code || useContext(CodeBlockContext).code;
```

Is it the same?

Reply:
Wow! Your suggestion is so much cleaner and beautiful, thanks for that gg <3
I'll refactor this block